isso/views/comments.py

```diff
@@ -1396,10 +1398,25 @@ def admin(self, env, req):
         order_by = req.args.get('order_by', 'created')
         asc = int(req.args.get('asc', 0))
         mode = int(req.args.get('mode', 2))
-        comments = self.comments.fetchall(mode=mode, page=page,
-                                          limit=page_size,
-                                          order_by=order_by,
-                                          asc=asc)
+        comment_url = req.args.get('comment_url', '')
+
+        # Search for a specific comment by URL
+        if comment_url:
+            try:
+                # Parse the comment URL to extract the comment ID from the fragment
+                parsed_url = urlparse(comment_url)
+                fragment = parsed_url.fragment
+                comment_id = int(fragment.split('-')[-1])
+
+                comments = self.comments.fetchall(comment_id=comment_id, limit=1)
+            except Exception:
```

Comment:
Please catch more specific exceptions and more specific to what's raising them; "Exception" is too broad and can catch exceptions we don't intend to.